EIPS/eip-7779.md

```diff
@@ -0,0 +1,106 @@
+
+## **Motivation**
+
+### **Predictable Gas Limit Growth**
+
+- **Current Issue:**
+  - The Ethereum network faces increasing demand, but changes to the gas limit are often manually adjusted by miners or validators based on their preferences, which may cause unpredictable block sizes and network performance issues.
+
+- **Need for Change:**
+  - A systematic and predictable increase of the gas limit will help scale the network while giving the ecosystem time to adjust to larger block sizes, without needing to rely on ad hoc decisions by network participants.
+
+### **Gradual Increase with Deactivation Safeguard**
+
+- **Controlled Growth:**
+  - Instead of sudden or unpredictable changes, this EIP proposes incremental gas limit increases over two years, which will gradually double the block capacity.
+  
+- **Automatic Deactivation:**
+  - A safeguard deactivation block will halt the increase after two years, ensuring that the gas limit does not grow indefinitely without further consensus from the community.
+
+## **Specification**
+
+### **Incremental Gas Limit Increase**
+
+1. **Initial Block Gas Limit:**
+   - At the start of the proposal, the block gas limit is set to 30 million gas.
+
+2. **Increase Rate:**
+   - The block gas limit will increase by 0.00475% per block (approximately 1.5% per day), such that after approximately 730 days (two years), the gas limit reaches 60 million.
+
+3. **Deactivation Block:**
+   - The gas limit increase will automatically halt at a specific block height, calculated as the block occurring roughly two years after the proposal's implementation. This "deactivation block" ensures no further increases occur after this period.
+
+4. **Formula for Gas Limit Increase:**
+   - The gas limit `GL_t` at block `t` is calculated as:
+   ```python
+   GL_t = GL_0 * (1 + r)^t
+   ```
+   Where:
+   - `GL_0` is the initial gas limit (30 million gas).
+   - `r` is the per-block increase rate (0.00475% or 0.0000475).
+   - `t` is the number of blocks since the implementation of the EIP.
+   - The increase halts once `t` reaches the block corresponding to two years (approximately 5,184,000 blocks, assuming 12-second block times).
+
+5. **Post-Deactivation Block Limit:**
+   - After the deactivation block, the gas limit remains fixed at 60 million gas unless further action is taken by the community.
+
+### **Implementation**
+
+- **Client Updates:**
+  - Ethereum clients must implement the automatic gas limit increase according to the specified rate and formula, with enforcement of the deactivation block.
+  
+- **Gas Limit Management:**
+  - Miners and validators are not required to adjust the gas limit manually during this period, as the increase will be automated at the protocol level.
```

Comment:
If miners decide to adjust the gas limit, this would yield an invalid block according to this EIP, so miners are unable to adjust the gas limit.

Reply:
Fixed your concerns in the updated version of the EIP